Server.py

```diff
@@ -107,7 +124,7 @@ def start(self):
             print(exc)
             print(traceback.format_exc())
             sys.exit(1)
-
+        client_socket.close()
```

Comment:
I dont think you need to explicitly call .close() here because the WITH statement will automatically close the socket.